api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -506,10 +506,10 @@ type HostedClusterSpec struct {
 	// Max is 6 to account for OIDC;OVNSbDb for backward compatibility though they are no-op.
 	//
 	// +kubebuilder:validation:MaxItems=6
-	// +kubebuilder:validation:MinItems=4
+	// +kubebuilder:validation:MinItems=3
 	// +kubebuilder:validation:ListType=atomic
 	// -kubebuilder:validation:XValidation:rule="self.all(s, !(s.service == 'APIServer' && s.servicePublishingStrategy.type == 'Route') || has(s.servicePublishingStrategy.route.hostname))",message="If serviceType is 'APIServer' and publishing strategy is 'Route', then hostname must be set"
-	// -kubebuilder:validation:XValidation:rule="['APIServer', 'OAuthServer', 'Konnectivity', 'Ignition'].all(requiredType, self.exists(s, s.service == requiredType))",message="Services list must contain at least 'APIServer', 'OAuthServer', 'Konnectivity', and 'Ignition' service types"
+	// -kubebuilder:validation:XValidation:rule="self.platform.type == "IBMCloud" ? ['APIServer', 'OAuthServer', 'Konnectivity'].all(requiredType, self.exists(s, s.service == requiredType))",message="Services list must contain at least 'APIServer', 'OAuthServer', and 'Konnectivity' service types" : ['APIServer', 'OAuthServer', 'Konnectivity', 'Ignition'].all(requiredType, self.exists(s, s.service == requiredType))",message="Services list must contain at least 'APIServer', 'OAuthServer', 'Konnectivity', and 'Ignition' service types"
```

Comment:
FWIW I just noticed this rules are actually commented out via "-" so the change is not strictly needed. See comment https://github.com/openshift/hypershift/pull/5782/files#diff-90f35474cefb9076bf5b33b62c1f9533575ed1d031538c824ba79ce6a45bf19fR515

Reply:
Oh gotcha, should I keep it there for now since it seems like the plan is to enable it eventually? No preference on my end here